kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/model/ingress/ClusterIPIngressDefinitionSniHostIdentifiesNodeTest.java

```diff
@@ -0,0 +1,371 @@
+    public static final String CLUSTER_NAME = "my-cluster";
+    public static final String PROXY_NAME = "my-proxy";
+    public static final String NAMESPACE = "my-namespace";
+
+    // @formatter:off
+    private static final KafkaProxyIngress INGRESS = new KafkaProxyIngressBuilder()
+            .withNewMetadata()
+                .withName(INGRESS_NAME)
+                .withNamespace(NAMESPACE)
+            .endMetadata()
+            .withNewSpec()
+                .withNewClusterIP()
+                    .withProtocol(ClusterIP.Protocol.TLS)
+                    .withNodeIdentifiedBy(ClusterIP.NodeIdentifiedBy.SNI)
+                .endClusterIP()
+            .endSpec()
+            .build();
+    // @formatter:on
+
+    public static final String CERTIFICATE_NAME = "certificate";
+    private static final VirtualKafkaCluster VIRTUAL_KAFKA_CLUSTER = new VirtualKafkaClusterBuilder()
+            .withNewMetadata()
+            .withName(CLUSTER_NAME)
+            .withNamespace(NAMESPACE)
+            .endMetadata()
+            .withNewSpec()
+            .addNewIngress()
+            .withNewIngressRef()
+            .withName(INGRESS_NAME)
+            .endIngressRef()
+            .withNewTls()
+            .withNewCertificateRef()
+            .withName(CERTIFICATE_NAME)
+            .endCertificateRef()
+            .endTls()
+            .endIngress()
+            .endSpec()
+            .build();
+    // @formatter:on
+    public static final String PROXY_UID = "my-proxy-uid";
+    public static final KafkaProxy PROXY = new KafkaProxyBuilder().withNewMetadata().withName(PROXY_NAME).withUid(PROXY_UID).withNamespace(NAMESPACE).endMetadata()
+            .build();
+    public static final NodeIdRanges NODE_ID_RANGE = createNodeIdRange("a", 1L, 3L);
+    public static final int SHARED_TLS_PORT = 9191;
+
+    @Test
+    void numIdentifyingPortsIsZero() {
+        ClusterIPIngressDefinition definition = new ClusterIPIngressDefinition(INGRESS, VIRTUAL_KAFKA_CLUSTER, PROXY,
+                List.of(NODE_ID_RANGE), null);
+        // 3 ports for the nodeId range + 1 for bootstrap
+        assertThat(definition.numIdentifyingPortsRequired()).isEqualTo(0);
+    }
+
+    @Test
+    void requiresASharedTlsPort() {
+        ClusterIPIngressDefinition definition = new ClusterIPIngressDefinition(INGRESS, VIRTUAL_KAFKA_CLUSTER, PROXY,
+                List.of(NODE_ID_RANGE), null);
+        // 3 ports for the nodeId range + 1 for bootstrap
+        assertThat(definition.requiresSharedTLSPort()).isTrue();
+    }
+
+    @Test
+    void createInstancesWithSharedPort() {
+        // given
+        ClusterIPIngressDefinition definition = new ClusterIPIngressDefinition(INGRESS, VIRTUAL_KAFKA_CLUSTER, PROXY,
+                List.of(createNodeIdRange("a", 1L, 3L)), null);
+        // when
+        // then
+        // expect to be given a shared TLS port
+        assertThat(definition.createIngressModel(null, null, SHARED_TLS_PORT)).isNotNull();
```

Comment:
we don't do anything to verify that we have a shared port here (I'm guessing its the fact that the 3rd arg is the shared port).